lamp.py

```diff
@@ -0,0 +1,209 @@
+        sys.exit(1)
+
+    if not os.access(install_dir, os.W_OK):
+        logging.critical(
+            'Please ensure you have write access to the parent directory for gennytoolchain: '
+            '`sudo chown $USER %s`', install_dir)
+        sys.exit(1)
+
+    toolchain_tarball = os.path.join(install_dir, 'gennytoolchain.tgz')
+    if os.path.isfile(toolchain_tarball):
+        logging.info('Skipping downloading %s', toolchain_tarball)
+    else:
+        logging.info('Downloading gennytoolchain, please wait...')
+        urllib.request.urlretrieve(url, toolchain_tarball)
+        logging.info('Finished Downloading gennytoolchain as %s', toolchain_tarball)
+
+    toolchain_dir = os.path.join(install_dir, 'gennytoolchain')
+    if os.path.exists(toolchain_dir):
+        logging.info('Skipping extracting the toolchain into: %s', toolchain_dir)
+    else:
+        logging.info('Extracting gennytoolchain, please wait...')
+        tarball = tarfile.open(toolchain_tarball)
+        tarball.extractall(toolchain_dir)
+        logging.info('Finished extracting gennytoolchain into %s', toolchain_dir)
+
+    return toolchain_dir
+
+
+def _create_compile_environment(toolchain_dir, triplet_os):
+    env = os.environ.copy()
+    paths = [env['PATH']]
+
+    # For mongodbtoolchain compiler (if there).
+    paths.insert(0, '/opt/mongodbtoolchain/v3/bin')
+
+    # For cmake and ctest
+    cmake_bin_relative_dir = {
+        'linux': 'downloads/tools/cmake-3.13.3-linux/cmake-3.13.3-Linux-x86_64/bin',
+        'osx': 'downloads/tools/cmake-3.13.3-osx/cmake-3.13.3-Darwin-x86_64/CMake.app/Contents/bin'
+    }[triplet_os]
+    paths.insert(0, os.path.join(toolchain_dir, cmake_bin_relative_dir))
+
+    # For ninja
+    ninja_bin_dir = os.path.join(toolchain_dir,
+                                 'downloads/tools/ninja-1.8.2-{}:'.format(triplet_os))
+    paths.insert(0, ninja_bin_dir)
+
+    env['PATH'] = ':'.join(paths)
+    env['NINJA_STATUS'] = '[%f/%t (%p) %es]'  # make the ninja output even nicer
+
+    logging.debug('Using environment: %s', env)
+    return env
+
+
+def cmake(toolchain_dir, cmdline_cmake_args, triplet_os, env):
+    cmake_cmd = ['cmake', '-B', 'build', '-G', 'Ninja']
+    # We set both the prefix path and the toolchain file here as a hack to allow cmake
+    # to find both shared and static libraries. vcpkg doesn't natively support a project
+    # using both.
+    cmake_prefix_path = os.path.join(toolchain_dir, 'installed/x64-{}-shared'.format(triplet_os))
+    cmake_toolchain_file = os.path.join(toolchain_dir, 'scripts/buildsystems/vcpkg.cmake')
+
+    cmake_cmd += [
+        '-DCMAKE_PREFIX_PATH={}'.format(cmake_prefix_path),
+        '-DCMAKE_TOOLCHAIN_FILE={}'.format(cmake_toolchain_file),
+        '-DVCPKG_TARGET_TRIPLET=x64-{}-static'.format(triplet_os),
+    ]
+
+    cmake_cmd += cmdline_cmake_args
+
+    logging.info('Running cmake: %s', cmake_cmd)
+    subprocess.run(cmake_cmd, env=env)
```

Comment:
I think you'll really want to specify an output file, subprocess can have issues with output buffer flood.